File: app/src/main/java/com/mrboomdev/awery/app/data/db/dao/RepositoryDao.java
package com.mrboomdev.awery.app.data.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.mrboomdev.awery.ext.source.Repository;

import java.util.List;

@Dao
public interface RepositoryDao {

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	void add(Repository url);

	@Query("SELECT * from repository WHERE manager = :manager")
	List<Repository> getRepositories(String manager);

	@Delete
	void remove(Repository url);
}

File: .old-shared/src/commonMain/kotlin/com/mrboomdev/awery/data/dao/RepositoryDao.kt
package com.mrboomdev.awery.data.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.mrboomdev.awery.data.entity.DBRepository

@Dao
interface RepositoryDao {
	@Insert(onConflict = OnConflictStrategy.REPLACE)
	suspend fun insert(repository: DBRepository)

	@Delete
	suspend fun delete(repository: DBRepository)
	
	@Query("SELECT * FROM DBRepository WHERE manager = :manager")
	suspend fun fromManager(manager: String): List<DBRepository>
}